File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        int orignal = x;
        long long reversed = 0;
        while(x > 0){
            int digit = x % 10;
            reversed = reversed * 10 + digit;
            x /= 10;
        }
        return orignal == reversed;
    }
};

File: 9-palindrome-number/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False

        orignal = x
        reverse = 0
        while x > 0:
            digit = x % 10
            reverse = reverse * 10 + digit
            x //= 10

        return orignal == reverse